src/app/pages/PerpetualPage/components/EditMarginDialog/index.tsx

```diff
@@ -0,0 +1,230 @@
+import { usePerpetual_accountBalance } from '../../hooks/usePerpetual_accountBalance';
+
+enum EditMarginDialogMode {
+  increase,
+  decrease,
+}
+
+export const EditMarginDialog: React.FC = () => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { pairType: currentPairType, modal, modalOptions } = useSelector(
+    selectPerpetualPage,
+  );
+  const ammState = usePerpetual_queryAmmState();
+  const perpParameters = usePerpetual_queryPerpParameters();
+  const traderState = usePerpetual_queryTraderState();
+  const trade = useMemo(
+    () => (isPerpetualTrade(modalOptions) ? modalOptions : undefined),
+    [modalOptions],
+  );
+  const pair = useMemo(
+    () => trade?.pairType && PerpetualPairDictionary.get(trade.pairType),
+    [trade],
+  );
+  const { available } = usePerpetual_accountBalance(
+    trade?.pairType || currentPairType,
+  );
+
+  const [mode, setMode] = useState(EditMarginDialogMode.increase);
+  const onSelectIncrease = useCallback(
+    () => setMode(EditMarginDialogMode.increase),
+    [],
+  );
+  const onSelectDecrease = useCallback(
+    () => setMode(EditMarginDialogMode.decrease),
+    [],
+  );
+
+  const [margin, setMargin] = useState('0');
+  const [changedTrade, setChangedTrade] = useState(trade);
+
+  const onClose = useCallback(
+    () => dispatch(actions.setModal(PerpetualPageModals.NONE)),
+    [dispatch],
+  );
+
+  const onSubmit = useCallback(
+    () =>
+      // TODO: implement review and excecution for EditMarginDialog
+      dispatch(
+        actions.setModal(PerpetualPageModals.TRADE_REVIEW, changedTrade),
+      ),
+    [dispatch, changedTrade],
+  );
+
+  const [maxAmount, maxAmountWei] = useMemo(() => {
+    if (mode === EditMarginDialogMode.increase) {
+      // Fees don't need to be subtracted, since Collateral is not paid with the Network Token
+      return [Number(fromWei(available)), available];
+    } else {
+      const requiredMargin =
+        (Math.abs(Number(fromWei(trade?.amount || '0'))) *
+          getBase2CollateralFX(ammState, true)) /
+        (pair?.config.leverage.max || 1);
+      const maxAmount = traderState.availableCashCC - requiredMargin;
+      return [maxAmount, toWei(maxAmount.toPrecision(8))];
+    }
+  }, [
+    mode,
+    available,
+    trade?.amount,
+    pair,
+    ammState,
+    traderState.availableCashCC,
+  ]);
+
+  const onChangeMargin = useCallback(
+    (margin?: string) =>
+      setMargin(currentMargin =>
+        Math.max(
+          0,
+          Math.min(maxAmount, margin ? Number(margin) : Number(currentMargin)),
+        ).toString(),
+      ),
+    [maxAmount],
+  );
+
+  const liquidationPrice = useMemo(
+    () =>
+      calculateApproxLiquidationPrice(
+        traderState,
+        ammState,
+        perpParameters,
+        changedTrade ? Number(fromWei(changedTrade.amount)) : 0,
+        (mode === EditMarginDialogMode.increase ? 1 : -1) * Number(margin),
```

Comment:
I see this is used also on line 132 so it can be extracted to a variable (e.g. signedMarginAmount), use `useMemo` as well.